velox/connectors/hive/HiveConnectorSplit.h

```diff
@@ -42,6 +42,7 @@ struct HiveConnectorSplit : public connector::ConnectorSplit {
   /// These represent columns like $file_size, $file_modified_time that are
   /// associated with the HiveSplit.
   std::unordered_map<std::string, std::string> infoColumns;
+  std::optional<std::string> rowIndexColumn;
```

Comment:
hey @Yuhta  you can see for `infoColumns` paramter which used for generateing const metadata column as `filesize` etc and `partitionKeys` both  actually comming from `ColumnHandle` with type  and `kSynthesized` and `kPartitionKey` and still been passed here.

We just leverage same way to pass `rowIndexColumn`, otherwise we need pass `ColumnHandle` itself, i think current way should be fine, since we already use it.

Reply:
`fileSize` and `partitionKeys` are good because they can be different from split to split.  `rowIndexColumn` should not be here because they are guaranteed to be the same.

Reply:
got it! make sense @Yuhta , let me see how to refactor this part.

Reply:
done